Source/EasyNetQ/Consumer/HandlerCollection.cs

```diff
@@ -61,7 +64,7 @@ public IHandlerRegistration Add<T>(Func<IMessage<T>, MessageReceivedInfo, Cancel
                 throw new EasyNetQException("No handler found for message type {0}", messageType.Name);
             }
 
-            return (m, i, c) => Task.CompletedTask;
+            return (m, i, c) => Task.FromResult(AckStrategies.Ack);
```

Comment:
Confused on this line - wouldn't this return a message handler that when invoked simply returns the Ack method? Would something then not have to call that method? Or should this be Task.FromResult(AckStrategies.Ack());

Reply:
AckStrategies.Ack is a reference to a method and it used to execute BasicAck on IModel. Also, I've added several overloads which could be used when no advanced control of acknowledgement of a message is needed.

Reply:
Took me a while to get my head around where that Ack method would be called, and I see it's where you used to always return Ack from the InvokeUserMessageHandlerInternalAsync method, but now it depends on what the specific handler decides 👍